src/View/ImageLoader.py

```diff
@@ -81,75 +85,82 @@ def load(self, interrupt_flag, progress_callback):
                 return False
 
             progress_callback.emit(("Getting contour data...", 30))
-            dict_raw_contour_data, dict_numpoints = \
-                ImageLoading.get_raw_contour_data(dataset_rtss)
+            dict_raw_contour_data, dict_numpoints = ImageLoading.get_raw_contour_data(
+                dataset_rtss
+            )
 
             # Determine which ROIs are one slice thick
             dict_thickness = ImageLoading.get_thickness_dict(
-                dataset_rtss, read_data_dict)
+                dataset_rtss, read_data_dict
+            )
 
             if interrupt_flag.is_set():  # Stop loading.
                 return False
 
             progress_callback.emit(("Getting pixel LUTs...", 50))
             dict_pixluts = ImageLoading.get_pixluts(read_data_dict)
-
+            progress_callback.emit(("Getting pixel LUTS...", 99))
             if interrupt_flag.is_set():  # Stop loading.
                 return False
 
+            progress_callback.emit(("Loading RTSS...", 1))
             # Add RTSS values to PatientDictContainer
             patient_dict_container.set("rois", rois)
             patient_dict_container.set("raw_contour", dict_raw_contour_data)
             patient_dict_container.set("num_points", dict_numpoints)
             patient_dict_container.set("pixluts", dict_pixluts)
+            progress_callback.emit(("Loading RTSS...", 100))
+            if "rtdose" not in file_names_dict:
+                progress_callback.emit(("Synthesizing RT Dose...", 1))
```

Comment:
this is an area that needs investigation, because it's pretty slow.
Having this visual display will help the user understand what's happening at the time, and will make performance evaluation easier when some optimisation is attempted.